web/app/js/components/Deployments.jsx

```diff
@@ -60,35 +68,90 @@ export default class Deployments extends React.Component {
     this.setState({ pendingRequests: true });
 
     let rollupPath = `${this.props.pathPrefix}/api/metrics?window=${this.state.metricsWindow}`;
-    let timeseriesPath = `${rollupPath}&timeseries=true`;
     let podPath = `${this.props.pathPrefix}/api/pods`;
     let rollupRequest = fetch(rollupPath).then(r => r.json());
-    let timeseriesRequest = fetch(timeseriesPath).then(r => r.json());
+
     let podRequest = fetch(podPath).then(r => r.json());
 
-    Promise.all([rollupRequest, timeseriesRequest, podRequest])
-      .then(([metrics, ts, p]) => {
-
-        let po = this.processDeploys(p.pods);
-        let m = _.compact(processMetrics(metrics.metrics, ts.metrics, "targetDeploy"));
-        let combinedMetrics = this.combineDeploymentsWithMetrics(po, m);
-        this.setState({
-          metrics: combinedMetrics,
-          lastUpdated: Date.now(),
-          pendingRequests: false,
-          loaded: true
-        });
-      }).catch(() => {
+    Promise.all([rollupRequest, podRequest])
+      .then(([rollup, p]) => {
+        let poByDeploy = this.getDeploymentList(p.pods);
+        let meshDeploys = _.compact(processRollupMetrics(rollup.metrics, "targetDeploy"));
+        let combinedMetrics = this.addDeploysWithNoMetrics(poByDeploy, meshDeploys);
+
+        this.loadTimeseriesFromServer(meshDeploys, combinedMetrics);
+      })
+      .catch(() => {
         this.setState({ pendingRequests: false });
       });
   }
 
-  renderPageContents() {
-    let leastHealthyDeployments = _(this.state.metrics)
-      .reject(m => m.added === false)
-      .sortBy(m => m.rollup.successRate)
-      .take(3)
+  loadTimeseriesFromServer(meshDeployMetrics, combinedMetrics) {
+    let limitSparklineData = _.size(meshDeployMetrics) > maxTsToFetch;
+    if (this.state.pendingTsRequests) {
+      return;
+    }
+    this.setState({ pendingTsRequests: true });
```

Comment:
Now that you're only setting state after all of the API calls have returned, I don't think you need to worry about setting and checking `pendingTsRequests` as part of this method. I'd probably just remove it.

Reply:
True, thanks!